### PR TITLE
Voting backend

### PR DESCRIPTION
**Pull request by VinaiRachakonda**

### CODE REVIEW

File: seeds/companies.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+exports.seed = function(knex) {
```
Comment (casey-chow) [inline seeds/companies.js]: Might be more useful to use [faker](
https://github.com/marak/Faker.js/) for this imo.

File: server/models/companies.model.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
+// See https://vincit.github.io/objection.js/#models
+// for more of what you can do here.
+import { Model } from 'objection';
+
+export default class Company extends Model {
+  static get tableName() {
+    return 'companies';
+  }
+
+  static get jsonSchema() {
```
Comment (casey-chow) [inline server/models/companies.model.js]: JSON schema is still here. Merge issue?

File: server/services/votes/whovoted.service.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/*
```
Comment (casey-chow) [inline server/services/votes/whovoted.service.js]: I think it makes more sense to add this as a field to a company, using the `alterItems` or `fastJoin` hook to tabulate the counts within the companies service. It reduces a round trip and the number of APIs we have to think about.

File: server/services/votes/finalize_votes.service.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
+import errors from '@feathersjs/errors';
+
+/*
+ * This service is used to determine for a given voteType and voteType who
+ * has submitted votes.
+ *
+ * For example when considering company_id = 1 and vote_type = 'Final' it will return all
+ * the names of the partners who submitted a vote.
+ *
+ * ex: PATCH /finalize_votes/1
```
Comment (casey-chow) [inline server/services/votes/finalize_votes.service.js]: `s/finalize_votes/finalize-votes/`

File: server/services/votes/finalize_votes.service.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
+import errors from '@feathersjs/errors';
```
Comment (casey-chow) [inline server/services/votes/finalize_votes.service.js:1]: Maybe rename to `votes-finalize` Or `companies/finalize-votes`? Not sure what the best API design here would be, not to mention that finalizing votes is

File: server/models/votes.model.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
+import { Model } from 'objection';
+
+export default class Vote extends Model {
+  static get tableName() {
```
Comment (casey-chow) [inline server/models/votes.model.js]: nit: `static tableName = ‘votes’;`

**Review by casey-chow — COMMENTED** [review]
I think it’d be good to prefix all the service names with `api/` since we’re planning on serving the client and API on the same server—prevents name collisions in the future.

File: server/models/votes.model.js (hunk 6 above)
Comment (VinaiRachakonda) [reply server/models/votes.model.js]: tableName is a feathers-objection supported method

File: server/services/votes/finalize_votes.service.js (hunk 5 above)
Comment (VinaiRachakonda) [reply server/services/votes/finalize_votes.service.js:1]: I got the concept of finalizing votes from the figma designs. Maybe the term update votes is a better idea? We just need a way to submit Pre-votes that can't be changed and then move to final votes that shouldn't be changed after they're "finalized"

File: server/services/votes/votes.service.js [hunk 7]
```diff
@@ -9,11 +9,23 @@ export default (app) => {
...
+          builder.select(['first_name', 'last_name']);
+        },
+      },
+    ],
   };
 
   // Initialize our service with any options it requires
-  app.use('/votes', createObjectionService(options));
-  app.service('votes').hooks(hooks);
+  app.use('api/votes', createObjectionService(options));
```
Comment (casey-chow) [inline server/services/votes/votes.service.js]: replace `api/votes` with `/api/votes`

File: server/services/votes/finalize_votes.service.js [hunk 8]
```diff
@@ -102,5 +102,5 @@ export default (app) => {
     },
   };
 
-  app.use('/finalize-votes', FinalizeVotesService);
+  app.use('api/votes/finalize', FinalizeVotesService);
```
Comment (casey-chow) [inline server/services/votes/finalize_votes.service.js]: replace `api/votes/finalize` with `/api/votes/finalize`

File: server/services/users/users.service.js [hunk 9]
```diff
@@ -10,6 +10,6 @@ export default (app) => {
     paginate,
   };
 
-  app.use('/users', createObjectionService(options));
+  app.use('users', createObjectionService(options));
```
Comment (casey-chow) [inline server/services/users/users.service.js]: replace `users` with `/users`

**Review by casey-chow — APPROVED** [review]

**Pull request merged**